File: Lab7/DBLabSkeleton/app/src/main/java/mobiledev/unb/ca/sqlitelab/DatabaseHelper.java
package mobiledev.unb.ca.sqlitelab;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class DatabaseHelper extends SQLiteOpenHelper {
    // Database information
    private static final String DATABASE_NAME = "SQLITE_LAB.DB";
    private static final int DATABASE_VERSION = 1;

    // Table columns
    public static final String _ID = "_id";
    public static final String ITEM = "item";
    public static final String NUM = "number";
    private static String[] COLUMNS = { _ID, ITEM, NUM };

    // Table creation statement
    public static final String TABLE_NAME = "mytable";
    private static final String TABLE_CREATE =
            "CREATE TABLE " + TABLE_NAME + " (" + _ID
                    + " INTEGER PRIMARY KEY AUTOINCREMENT, "
                    + ITEM + " TEXT, " + NUM + " INT);";

    DatabaseHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(TABLE_CREATE);
    }

    @Override
    public void onUpgrade (SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_NAME);
    }
}

File: Winter2022/examples/kotlin/Lecture 7/DBDemo/app/src/main/java/mobiledev/unb/ca/sqlitetest/DatabaseHelper.kt
package mobiledev.unb.ca.sqlitetest

import android.content.Context
import android.database.sqlite.SQLiteOpenHelper
import android.database.sqlite.SQLiteDatabase

class DatabaseHelper internal constructor(context: Context?) :
    SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(TABLE_CREATE)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL("DROP TABLE IF EXISTS $TABLE_NAME")
    }

    companion object {
        // Database information
        private const val DATABASE_NAME = "ITEM_TEST.DB"
        private const val DATABASE_VERSION = 1

        // Table columns
        const val ITEM_ID = "_id"
        const val ITEM = "item"
        const val NUM = "number"

        @JvmField
        var COLUMNS = arrayOf(ITEM_ID, ITEM, NUM)

        // Table creation statement
        const val TABLE_NAME = "mytable"
        private const val TABLE_CREATE = ("CREATE TABLE " + TABLE_NAME + " (" + ITEM_ID
                + " INTEGER PRIMARY KEY AUTOINCREMENT, "
                + ITEM + " TEXT, " + NUM + " INT);")
    }
}